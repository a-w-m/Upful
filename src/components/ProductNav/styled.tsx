import styled from "styled-components";
import device from "../Layout/mediaQuery"
export const Wrapper = styled.nav`
display: flex;
gap: 1rem;
font-size: 1rem;
justify-content: flex-start;
align-items: end;
width: 100%;
height: 1.8rem;


@media ${device.laptop}{
    grid-area: nav;
}

& > a{
    font-size: 1rem;
    text-decoration: none;
    color: var(--main-color);
    text-transform: uppercase;
}
& >a:hover
{
        color: var(--header-nav-link-hover)
    }




& > svg{
    height:1.2rem;
    width: .8rem;
}

& > span {

    text-transform: uppercase;
    vertical-align:baseline;

}

`