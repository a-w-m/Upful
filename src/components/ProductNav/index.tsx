import React from 'react'
import {Wrapper} from "./styled"
import {Link} from 'gatsby'
import {ChevronRight} from "../Icons/index"

interface Props{
    collection: string
    title: string
}

const ProductNav: React.FC<Props> = (props)=>{

    const {collection, title} = props
return(
    <Wrapper>
        <Link to = {"/index"}>
            HOME
        </Link>
        <ChevronRight></ChevronRight>
        <Link to ={`/${collection}`}>
            {collection}
        </Link>
        <ChevronRight></ChevronRight>
        <span>{title}</span>
    </Wrapper>
)
}

export default ProductNav