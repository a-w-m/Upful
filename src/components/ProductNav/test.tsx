import React from 'react'
import {getAllByRole, render, screen} from "@testing-library/react"
import ProductNav from "."

describe("ProductNav", ()=>{

    beforeEach(()=>{
        render(<ProductNav {...mockData}/>)
    })
    //snapshot
    it("should match snapshot", ()=>{
       const container =  screen.getByRole('navigation')
       expect(container).toMatchSnapshot()
    })
    it("should link to home page",()=>{
    const home = screen.getByRole('link', {name: /home/i})
    expect(home).toHaveAttribute('href', '/')
    })

    it("should link to collection", ()=>{
        const collection = screen.getByRole('link', {name: `${mockData.collection}`})
        expect(collection).toHaveAttribute('href', `/${mockData.collection}`)
    })

    it("should display title", ()=>{
        const title = screen.getByText(`${mockData.title}`)
        expect(title).toBeInTheDocument()
    })
    
    it ("should display two right chevrons", ()=>{
        const chevrons = screen.getAllByRole('img', {hidden: true})
        expect(chevrons).toHaveLength(2)
    })
})

const mockData = {
    title: 'Product',
    collection: 'Test'
}