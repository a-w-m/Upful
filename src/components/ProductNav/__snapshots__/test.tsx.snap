// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNav should match snapshot 1`] = `
<nav
  class="sc-bczRLJ eoUIHq"
>
  <a
    href="/"
  >
    HOME
  </a>
  <svg
    aria-label="chevron-right"
    class="bi bi-chevron-right"
    fill="currentColor"
    height="14"
    role="img"
    viewBox="0 0 16 16"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
      fill-rule="evenodd"
    />
  </svg>
  <a
    href="/Test"
  >
    Test
  </a>
  <svg
    aria-label="chevron-right"
    class="bi bi-chevron-right"
    fill="currentColor"
    height="14"
    role="img"
    viewBox="0 0 16 16"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
      fill-rule="evenodd"
    />
  </svg>
  <span>
    Product
  </span>
</nav>
`;
